cmd/kubelet/app/options/options.go

```diff
@@ -540,6 +540,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.StringSliceVar(&c.EnforceNodeAllocatable, "enforce-node-allocatable", c.EnforceNodeAllocatable, "A comma separated list of levels of node allocatable enforcement to be enforced by kubelet. Acceptable options are 'none', 'pods', 'system-reserved', and 'kube-reserved'. If the latter two options are specified, '--system-reserved-cgroup' and '--kube-reserved-cgroup' must also be set, respectively. If 'none' is specified, no additional options should be set. See https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/ for more details.")
 	fs.StringVar(&c.SystemReservedCgroup, "system-reserved-cgroup", c.SystemReservedCgroup, "Absolute name of the top level cgroup that is used to manage non-kubernetes components for which compute resources were reserved via '--system-reserved' flag. Ex. '/system-reserved'. [default='']")
 	fs.StringVar(&c.KubeReservedCgroup, "kube-reserved-cgroup", c.KubeReservedCgroup, "Absolute name of the top level cgroup that is used to manage kubernetes components for which compute resources were reserved via '--kube-reserved' flag. Ex. '/kube-reserved'. [default='']")
+	fs.StringVar(&c.Logging.Format, "logging-format", c.Logging.Format, `Sets the log format. Permitted formats: "text", "json".\nNon-default formats don't honor these flags: -add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --skip_headers, --skip_log_headers, --stderrthreshold, --log-flush-frequency.\nNon-default choices are currently alpha and subject to change without warning.`)
```

Comment:
will we be duplicating this message across all places we add --logging-format? can we do this in one place so we don't have to maintain the full list of valid formats, and the full list of flags that can't be used in combination with this in multiple places?

Reply:
can we expose/use unsupportedLoggingFlags to avoid building this manually, or move the flag registration methdo to staging/src/k8s.io/component-base/logs so we can use that private method

Reply:
This list of flags haven't changed for very long time. I don't think we are worried about this getting out of date. Also we should not be worried about a having duplicate too much. I don't think it makes sense to try to optimize current code for future.

I would love to refactor all possible duplicates of this flag, but we are still implementing validation of unsupported flags. Even thou there is 5 contributors working on this issue (with 5 PRs), only one can work at the time. It would really help if we could agree on one set of requirements allowing us to merge all PRs at once. 

PRs touching flags:
* https://github.com/kubernetes/kubernetes/pull/91501
* https://github.com/kubernetes/kubernetes/pull/91521
* https://github.com/kubernetes/kubernetes/pull/91532
* https://github.com/kubernetes/kubernetes/pull/91522
* https://github.com/kubernetes/kubernetes/pull/92394

Reply:
it doesn't need to block this PR, but can you add an issue or item to the list of follow-ups to unify this so we don't end up with a half-dozen copies of the same code that drift from each other?